spec/miq_ae_event_spec.rb

```diff
@@ -63,24 +62,65 @@
                           )
       end
 
-      context "with user owned VM" do
-        let(:vm_owner) { user }
+      shared_context "variables" do
+        let(:vm_owner)   { user }
+        let(:miq_server) { EvmSpecHelper.local_miq_server(:is_master => true) }
+        let(:args) do
+          { :miq_group_id => group.id, :tenant_id => tenant.id, :user_id => user.id}
+        end
+        let(:inputs) do
+          {:vm => vm}
+        end
+      end
 
-        it "has tenant" do
-          args = {:miq_group_id => group.id, :tenant_id => tenant.id, :user_id => user.id}
-          expect(MiqAeEngine).to receive(:deliver_queue).with(hash_including(args), anything)
+      shared_examples_for "#pass zone to raise_evm_event" do
+        include_context "variables"
 
-          MiqAeEvent.raise_evm_event("vm_create", vm, :vm => vm)
+        it "runs successfully" do
+          expect(MiqAeEngine).to receive(:deliver_queue).with(hash_including(args), hash_including(:zone => zone_name))
+          MiqAeEvent.raise_evm_event("vm_create", vm, inputs, options)
         end
       end
 
+      shared_examples_for "#zone not passed to raise_evm_event" do
+        include_context "variables"
+
+        it "runs successfully" do
+          expect(MiqAeEngine).to receive(:deliver_queue).with(hash_including(args), hash_excluding(:zone))
+          MiqAeEvent.raise_evm_event("vm_create", vm, inputs, options)
+        end
+      end
+
+      context "with zone" do
+        let(:zone) { FactoryGirl.create(:zone) }
+        let(:zone_name) { zone.name }
+        let(:options) do
+          {:zone => zone.name}
+        end
+        it_behaves_like "#pass zone to raise_evm_event"
+      end
+
+      context "nil zone" do
+        let(:zone_name) { nil }
+        let(:options) do
+          {:zone => nil}
+        end
+      end
+
+      context "nil options" do
+        let(:zone_name) { 'default' }
+        let(:options) do
+          {}
+        end
+        it_behaves_like "#zone not passed to raise_evm_event"
+      end
+
       context "with group owned VM" do
         let(:vm_group) { FactoryGirl.create(:miq_group, :tenant => FactoryGirl.create(:tenant)) }
 
         it "has tenant" do
           args = {:user_id => admin.id, :miq_group_id => vm_group.id, :tenant_id => vm_group.tenant.id}
           expect(MiqAeEngine).to receive(:deliver_queue).with(hash_including(args), anything)
-
```

Comment:
@tinaafitz No sure if you intentionally remove this line but it seems random with the other changes in this PR and now makes this test inconsistent with (most) of the remaining tests that separate the `expect().to receive` and the following method call.  I tend to prefer this separation between testing logic and the code being run for the test.